src/ChannelServer/Network/ChannelPacketSender.cs

```diff
@@ -1676,6 +1730,30 @@ public static void ZC_UPDATE_SP(Character character, short spValue)
 		}
 
 		/// <summary>
+		/// Updates a characters HP for damage and healing.
+		/// </summary>
+		/// <param name="character"></param>
+		/// <param name="amount"></param>
+		/// <param name="isDamage"></param>
+		public static void ZC_ADD_HP(Character character, int amount, bool isDamage)
+		{
+			character.HPChangeCounter += 1;
```

Comment:
Is it a good idea to do logic inside a packet sender? It seems like this might have unforeseen consequences if someone doesn't expect a sender to actually *apply* the damage, instead of just updating the client. I believe this would be better placed in a method on Character, which then calls this function.